packages/hoppscotch-app/components/environments/teams/Details.vue

```diff
@@ -0,0 +1,312 @@
+          :label="`${t('action.cancel')}`"
+          @click.native="hideModal"
+        />
+      </span>
+    </template>
+  </SmartModal>
+</template>
+
+<script setup lang="ts">
+import { computed, ref, watch } from "@nuxtjs/composition-api"
+import * as E from "fp-ts/Either"
+import * as A from "fp-ts/Array"
+import * as O from "fp-ts/Option"
+import * as TE from "fp-ts/TaskEither"
+import { flow, pipe } from "fp-ts/function"
+import { parseTemplateStringE } from "@hoppscotch/data"
+import { refAutoReset } from "@vueuse/core"
+import { clone } from "lodash"
+import { useI18n, useToast } from "~/helpers/utils/composables"
+import {
+  createTeamEnvironment,
+  updateTeamEnvironment,
+} from "~/helpers/backend/mutations/TeamEnvironment"
+import { GQLError } from "~/helpers/backend/GQLClient"
+import { TeamEnvironment } from "~/helpers/teams/TeamEnvironment"
+
+type EnvironmentVariable = {
+  id: number
+  env: {
+    key: string
+    value: string
+  }
+}
+
+const t = useI18n()
+const toast = useToast()
+
+const props = withDefaults(
+  defineProps<{
+    show: boolean
+    action: "edit" | "new"
+    editingEnvironment: TeamEnvironment | null
+    editingTeamId: string | undefined
+  }>(),
+  {
+    show: false,
+    action: "edit",
+    editingEnvironment: null,
+    editingTeamId: "",
+  }
+)
+
+const emit = defineEmits<{
+  (e: "hide-modal"): void
+}>()
+
+const idTicker = ref(0)
+
+const name = ref<string | null>(null)
+const vars = ref<EnvironmentVariable[]>([
+  { id: idTicker.value++, env: { key: "", value: "" } },
+])
+
+const clearIcon = refAutoReset<"trash-2" | "check">("trash-2", 1000)
+
+const evnExpandError = computed(() => {
+  const variables = pipe(
+    vars.value,
+    A.map((e) => e.env)
+  )
+
+  return pipe(
+    variables,
+    A.exists(({ value }) => E.isLeft(parseTemplateStringE(value, variables)))
+  )
+})
+
+const liveEnvs = computed(() => {
+  if (evnExpandError) {
+    return []
+  } else {
+    return [...vars.value.map((x) => ({ ...x, source: name.value! }))]
+  }
+})
+
+watch(
+  () => props.show,
+  (show) => {
+    if (show) {
+      if (props.editingEnvironment === null) {
+        name.value = null
+        vars.value = []
+      } else {
+        name.value = props.editingEnvironment.name ?? null
+        vars.value = pipe(
+          JSON.parse(props.editingEnvironment.variables) ?? [],
+          A.map((e: { key: string; value: string }) => ({
+            id: idTicker.value++,
+            env: clone(e),
+          }))
+        )
+      }
+    }
+  }
+)
+
+const clearContent = () => {
+  vars.value = []
+  clearIcon.value = "check"
+  toast.success(`${t("state.cleared")}`)
+  // TODO: DeleteAllVariablesFromTeamEnvironment
+}
+
+const addEnvironmentVariable = () => {
+  vars.value.push({
+    id: idTicker.value++,
+    env: {
+      key: "",
+      value: "",
+    },
+  })
+}
+
+const removeEnvironmentVariable = (index: number) => {
+  vars.value.splice(index, 1)
+}
+
+const saveEnvironment = () => {
```

Comment:
Save Environment Modal should wait until the actual saving to be done before the modal is closed along with a loading state.
It isn't currently.